langchain/agents/tools/json.ts

```diff
@@ -0,0 +1,101 @@
+import {Tool, Toolkit} from "./base";
+
+function parseInput(input: string): (string | number)[] {
+    const regex = /\[.*?]/g;
+    const _res = input.match(regex) ?? [];
+    const res = _res.map((i) => i.slice(1, -1).replace(/['"]+/g, ''));
+    return res.map((i) => (Number.isNaN(+i) ? i : +i));
+}
+
+export class JsonSpec {
+    obj: { [key: string]: any };
```

Comment:
any ideas on a better type here? Linter complaining about using any

Reply:
courtesy of supabase 

```
export type Json =
  | string
  | number
  | boolean
  | null
  | { [key: string]: Json }
  | Json[]

export type JsonObject = { [key: string]: Json };

export class JsonSpec {
    obj: JsonObject
    ...
}
```

Reply:
Awesome, thanks for this!